File: distributed_systems/exercise_1/Java-Exercise 2/src/Book.java

public class Book extends Publication {
	String author;
	long isbn;
	
	public Book(String title, String language, double price, String author, long isbn) {
		super(title, language, price);
		this.author = author;
		this.isbn = isbn;		
	}
	
	void print() {
		super.print();
		System.out.println(String.format("%s %d", this.author, this.isbn));
	}
}

File: objektorientierte_programmierung_grundlagen/aufgabenblatt_3/aufgabe_2/Book.cpp
#include <iostream>

#include "Book.h"

using namespace std;

Book::Book(string author, string title, long int isbn, double price) {
    this->author = author;
    this->title = title;
    this->isbn = isbn;
    this->price = price;
    this->id = ++Book::id_pool;
}

Book::~Book() {}

int Book::id_pool = 0;

void Book::print() {
    cout << "author: " << author << endl;
    cout << "title: " << title << endl;
    cout << "isbn: " << isbn << endl;
    cout << "price: " << price << endl;
    cout << "id: " << id << endl;
}
